store.rb

```diff
@@ -16,6 +16,18 @@ def initialize(opts)
     end
 
     @zk = nil
+
+    # cache related
```

Comment:
i'm not a fan of the amount of state here; there are 9 additional variables. i'm thinking about the number of ways to reduce that